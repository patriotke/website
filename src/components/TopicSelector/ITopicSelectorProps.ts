export type ITopicOption = {
  label: string;
  value: string;
};
export type ITopicSelectorProps = {
  placeholder: string;
  label?: string;
  value?: string;
  onChange?: (value: string) => void;
  options: ITopicOption[];
};
