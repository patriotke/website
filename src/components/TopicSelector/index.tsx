/**
 *
 * TopicSelector
 *
 */

'use client';

import { ReactElement } from 'react';
import {
  Select,
  SelectContent,
  SelectGroup,
  SelectItem,
  SelectLabel,
  SelectTrigger,
  SelectValue,
} from 'components/ui/select';
import { ITopicSelectorProps } from 'components/TopicSelector/ITopicSelectorProps';

function TopicSelector({
  placeholder,
  label,
  value,
  onChange,
  options = [],
}: ITopicSelectorProps): ReactElement {
  return (
    <Select value={value} onValueChange={onChange}>
      <SelectTrigger className="w-[180px]">
        <SelectValue placeholder={placeholder} />
      </SelectTrigger>
      <SelectContent>
        <SelectGroup>
          {label && <SelectLabel>{label}</SelectLabel>}
          {options.map((option) => (
            <SelectItem key={option.value} value={option.value}>
              {option.label}
            </SelectItem>
          ))}
        </SelectGroup>
      </SelectContent>
    </Select>
  );
}

export default TopicSelector;
