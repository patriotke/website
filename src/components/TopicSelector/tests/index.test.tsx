/* eslint-disable import/no-extraneous-dependencies */
import '@testing-library/jest-dom';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { render, screen } from '@testing-library/react';
import TopicSelector from '../index';

describe('TopicSelector', () => {
  it('renders TopicSelector', () => {
    render(<TopicSelector />);

    const { container } = render(<TopicSelector />);
    expect(container).toMatchSnapshot();
  });
});
