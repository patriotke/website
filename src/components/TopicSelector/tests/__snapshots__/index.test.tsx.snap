// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicSelector renders TopicSelector 1`] = `
<div>
  <div>
    TopicSelector
  </div>
</div>
`;
